### DISCUSSION TITLE
background image not showing

### DISCUSSION THREAD

**Discussion by jbuoncri1** [tailwindlabs/tailwindcss#14790] [Help] [answered]
Hello Everyone,

I'm trying to get get a background image to load via arbitrary values but the images aren't displaying. For context, I'm using the react-slick library and want to display a background image on each slide. I checked the path countless times and I'm pretty sure it's correct. Here's a link to the repo:
https://github.com/jbuoncri1/pokemart-nw

From `pages/home/index.tsx`:
<img width="523" alt="Screenshot 2024-10-24 at 13 14 06" src="https://github.com/user-attachments/assets/fc11d9e0-811f-4ebe-b6a0-c6fa8af34788">

What the page is displaying:
<img width="1675" alt="Screenshot 2024-10-24 at 13 14 58" src="https://github.com/user-attachments/assets/f9086b06-5a42-4319-9fb1-968128de3827">

Appreciate the help!

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

Instead, you could consider using the `style` attribute like:

```diff
 <div
   key={banner.id}
-  className={`bg-[url('${banner.image}')] w-full h-48`}
+  className="w-full h-48"
+  style={{ backgroundImage: `url('${banner.image}')` }}
 >

**Reply by jbuoncri1** [reply]
Shoot the style prop didn't work. Theimage  URL isn't showing in the DOM either

<img width="1680" alt="Screenshot 2024-10-24 at 13 26 19" src="https://github.com/user-attachments/assets/1d44759a-a093-4569-a1d2-bfae0eb7ff35">

**Reply by wongjn** [reply, marked as answer]
Seems like React Slick overrides the `style` attribute for the root `<div>`. You could consider using a child `<div>`:

```diff
 <div className="slider-container">
   <Slider {...settings}>
     {banners.map((banner) => (
-      <div
-        key={banner.id}
-        className={`bg-[url('${banner.image}')] w-full h-48`}
-      >
-        <p className="text-white">{banner.name}</p>
+      <div key={banner.id}>
+        <div
+          className="w-full h-48"
+          style={{ backgroundImage: `url('${banner.image}')` }}
+        >
+          <p className="text-white">{banner.name}</p>
+        </div>
       </div>
     ))}
   </Slider>
```

**Reply by jbuoncri1** [reply]
Appreciate it!
